embedded/drivers/src/gpio.cpp

```diff
@@ -0,0 +1,152 @@
+#include "gpio.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <iostream>
+#define NUM_GPIO_PINS 32  // dummy number right now
```

Comment:
So this might be tricky. The GPIO numbering is not very consistent. We will be using pin numbers as low as 27 and as high as 117. Let's just use 117 as an inclusive bound I guess